src/runtime/vulkan/vulkan.cc

```diff
@@ -496,27 +496,30 @@ VulkanDeviceAPI::VulkanDeviceAPI() {
     vkGetPhysicalDeviceQueueFamilyProperties(phy_dev, &queue_prop_count,
                                              dmlc::BeginPtr(queue_props));
     uint32_t queue_family_index = 0;
-    std::vector<VkDeviceQueueCreateInfo> queue_create_info;
-    float priority = 1.0f;
-    for (uint32_t i = 0; i < queue_props.size(); i++) {
-      // find queues that support compute
-      if (VK_QUEUE_COMPUTE_BIT & queue_props[i].queueFlags) {
-        VkDeviceQueueCreateInfo info;
-        info.sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
-        info.pNext = nullptr;
-        info.flags = 0;
-        info.queueFamilyIndex = i;
-        info.queueCount = 1;
-        info.pQueuePriorities = &priority;
-
-        queue_create_info.push_back(info);
-        // only use the first available queue for now
-        if (queue_create_info.size() == 0) {
-          queue_family_index = i;
-        }
+    float priority = 0.0f;
+    auto compute_dedicated = std::find_if(queue_props.begin(), queue_props.end(), [](auto prop) {
```

Comment:
Please add a comment indicating that this is for to find the compute decidated queue

Reply:
Perhaps it is better to put it into a sub member function, e.g. FindComputeQueue

Reply:
It is still better to find all queues. perhaps putting compute only queue first

Reply:
I refactored the compute queue finding code out. What did you mean in the last comment? This implementation attempts to find a compute-only queue first, and if no such queue exists, it will fall back to any queue that supports compute.

Reply:
I actually meant to still return a `std::vector<>` of all queues available, instead of a single queue